overlord/snapstate/backend/link.go

```diff
@@ -68,6 +68,13 @@ func (b Backend) LinkSnap(info *snap.Info, model *asserts.Model) error {
 		return err
 	}
 
+	// fontconfig is only relevant on classic
+	if release.OnClassic {
+		if err := updateFontconfigCaches(); err != nil {
```

Comment:
my only concern is that this is not an obvious place to expect this, adding a task for it is overkill/problematic tough. What's the speed in the case the cache is not there?  could this be put under an Ensure task?

Reply:
Yeah, I'm not happy about this particular place. I wanted it to be part of install/refresh and the other places looked worse. But your idea is also interesting:

Doing it in an Ensure() in the background probably also works - on my i5/ssd system it takes ~30s on an empty cache. Anecdotally I hear it can take minutes with many fonts on hdd systems. So we could run it in the background as part of snapd startup. Ideally we would synchronize with install somehow, i.e. if you "snap install vscode" right after login and the fontcache is generated in the background we would not consider the install finished until fontconfig is ready. But maybe I'm overthinking things here. Feedback/ideas welcome.

Reply:
I was expecting this to run only for *classic* snaps, not for all classic systems. Is there any way said caches are used in strictly confined snaps today?

Reply:
afaict yes, look at the desktop interface code, especially MountConnectedPlug

Reply:
Yeah, we could target it further for classic and desktop/unity7 snaps.